core/src/main/java/org/apache/iceberg/ManifestReader.java

```diff
@@ -268,8 +268,7 @@ CloseableIterable<ManifestEntry<F>> liveEntries() {
   /** @return an Iterator of DataFile. Makes defensive copies of files before returning */
   @Override
   public CloseableIterator<F> iterator() {
-    return CloseableIterable.transform(
-            liveEntries(), e -> e.file().copy(!dropStats(rowFilter, columns)))
+    return CloseableIterable.transform(liveEntries(), e -> e.file().copy(!dropStats(columns)))
```

Comment:
I realize that this isn't changing what's already done, but I think it would be better to have a variable for the result of `dropStats`. That's not going to change and we don't need to evaluate whether to drop stats for the columns every time this runs!